### PR TITLE
Remove support for obsolete framework versions

### PR DESCRIPTION
**Pull request by karolz-ms**
- Removes support for .NET 4.5.2 and .NET Core 2.1. The oldest frameworks explicitly targeted are .NET 4.6.2 and .NET Core 3.1
- Enables testing against .NET 6.0
- Updates some dependencies (Application Insights, Newtonsoft JSON, and some system packages)

Fixes issue https://github.com/Azure/diagnostics-eventflow/issues/399

### CODE REVIEW

File: src/Microsoft.Diagnostics.EventFlow.Core/Implementations/UtcMidnightNotifier.cs [hunk 1]
```diff
@@ -35,11 +32,6 @@ public static UtcMidnightNotifier Instance
         private UtcMidnightNotifier()
         {
             CreateNewTimer();
-
-            // TODO: Find an event to subscribe for .NET Core application when system time is changed.
-#if NET452
-            SystemEvents.TimeChanged += OnSystemTimeChanged;
```
Comment (karolz-ms) [inline src/Microsoft.Diagnostics.EventFlow.Core/Implementations/UtcMidnightNotifier.cs:41]: According to documentation, this API (`SystemEvents.TimeChanged`) relies on Windows message loop, so I do not think it ever worked properly in server environments. But nobody ever complained ¯\\_(ツ)_/¯

File: test/Microsoft.Diagnostics.EventFlow.FunctionalTests.HealthReporterBuster/Microsoft.Diagnostics.EventFlow.FunctionalTests.HealthReporterBuster.csproj [hunk 2]
```diff
@@ -1,129 +1,17 @@
...
-  </ItemGroup>
-  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
-  <!-- To modify your build process, add your task inside one of the targets below and uncomment it. 
-       Other similar extension points exist, see Microsoft.Common.targets.
-  <Target Name="BeforeBuild">
-  </Target>
-  <Target Name="AfterBuild">
-  </Target>
-  -->
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (xiaomi7732) [inline test/Microsoft.Diagnostics.EventFlow.FunctionalTests.HealthReporterBuster/Microsoft.Diagnostics.EventFlow.FunctionalTests.HealthReporterBuster.csproj:1]: Like this a lot!

File: test/Microsoft.Diagnostics.EventFlow.Inputs.Tests/ActivitySourceInputTests.cs [hunk 3]
```diff
@@ -3,7 +3,7 @@
 //  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
 // ------------------------------------------------------------
 
-#if NET5_0 || NETCOREAPP2_1 || NETCOREAPP3_1
+#if NET5_0 || NET5_0 || NETCOREAPP2_1 || NETCOREAPP3_1
```
Comment (xiaomi7732) [inline test/Microsoft.Diagnostics.EventFlow.Inputs.Tests/ActivitySourceInputTests.cs:6]: Is this a typo? Maybe `NET6_0`?

**Review by xiaomi7732 — APPROVED** [review]
Looks good to me. Just 1 minor comment, please kindly address before merge.

File: test/Microsoft.Diagnostics.EventFlow.Inputs.Tests/ActivitySourceInputTests.cs (hunk 3 above)
Comment (karolz-ms) [reply test/Microsoft.Diagnostics.EventFlow.Inputs.Tests/ActivitySourceInputTests.cs]: Yes, that was a typo. Fixed by dropping testing .NET 5.0, which will go out of support in early May 2022

**Comment by karolz-ms** [comment]
@xiaomi7732 just FYI there is still some changes coming. I had to make much more changes related to new .NET 6.0 tooling than I expected; I see one new, ETW-related test failure, and I might need to rev up some dependencies, not to mention rolling build still failing. :-/

**Comment by xiaomi7732** [comment]
Understood. I like where it is going. I am a bit sad that we have to deal with all these dets every once a while. Give me a ping when the code is ready for review again 👍

**Comment by karolz-ms** [comment]
@xiaomi7732 thanks much! Fortunately all the changes that I made after your review were just small test changes, dependency updates, and build process changes. So I am going to merge this and publish the resulting new Nuget packages sometime later this week.

**Comment by xiaomi7732** [comment]
Works for me! Thanks for the info!

**Pull request merged**